### PR TITLE
Feat/add aws scaling

### PR DESCRIPTION
**Pull request by ForsbergPhilip**
Created bash script for autoscaling AWS instance.

I have added a few `exit 1` because I want to try this first.

I have added restart: always to frontend and backend in docker-compose. I need to try if this restarts the containers.

### CODE REVIEW

File: auto-scale-aws.sh [hunk 1]
```diff
@@ -0,0 +1,90 @@
+#!/bin/bash
+
+# Restarts AWS instance and changes the volume
```
Comment (indrif) [inline auto-scale-aws.sh:3]: Ambiguos use of the word "volume". Volume in aws means disk so we should change this to "instance type". Same below.

File: auto-scale-aws.sh [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+# Using this function to remove double quotes from strings
+remove_double_quotes() {
+    echo $(sed -e 's/^"//' -e 's/"$//' <<<"$1")
+}
+
+off_season="Off season"
+instance_type_micro="t2.micro"
+instance_type_small="t2.small"
+instance_id="i-0ae22e94380789482"
```
Comment (indrif) [inline auto-scale-aws.sh:16]: Is it ok to expose instance id? Should be ok but to be safe we can expose it through github secrets instead

File: auto-scale-aws.sh [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+# Get instance volume
+instance_info=$(aws ec2 describe-instances --instance-ids  $instance_id --query "Reservations[*].Instances[*].{Status:State.Name,Instancetype:InstanceType}")
+instance_type=$(jq '.[0][0].Instancetype' <<< $instance_info)
+instance_type=$(remove_double_quotes "$instance_type")
+echo "InstanceType is $instance_type"
+
+
+# Fetch current_seanson. 
+response=$(curl http://diamonds.etimo.se/api/seasons/current)
```
Comment (indrif) [inline auto-scale-aws.sh:27]: Nice way to find out season! Didnt even think about using our own api myself :D

File: auto-scale-aws.sh [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+current_season=$(jq '.data.name'  <<< $response)
+current_season=$(remove_double_quotes "$current_season")
+
+if [ "$current_season" = "$off_season" ]; then
```
Comment (indrif) [inline auto-scale-aws.sh:31]: Shouldnt it be double equality here? `==`

File: auto-scale-aws.sh [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    new_instance_type=$instance_type_micro
+else
+    new_instance_type=$instance_type_small
+fi
+
+echo "InstanceType should be $new_instance_type"
+
+# Exit if the type is the same
+if [ "$instance_type" = "$new_instance_type" ]; then
```
Comment (indrif) [inline auto-scale-aws.sh:40]: Double equality?

File: auto-scale-aws.sh [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+status=$(jq '.[0][0].Status' <<< $instance_info)
+status=$(remove_double_quotes "$status")
+
+
+echo "Instance status: $status"
+# Just testing - exiting now
+exit 1;
+
+#Is the instance still running?
+if [ "$status" = "running" ]; then
```
Comment (indrif) [inline auto-scale-aws.sh:71]: same here?

File: auto-scale-aws.sh [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+
+# Not checking if volume is correct because we need to start the instance again anyway!
+
+# Start instance
+aws ec2 start-instances --instance-ids $instance_id
+echo "Instance started again"
+
+
+#Start docker containers! Need to try this!
+https://stackoverflow.com/questions/58141265/how-to-auto-restart-a-docker-compose-cluster-when-ec2-instance-reboots
```
Comment (indrif) [inline auto-scale-aws.sh:90]: Forgot comment on line

File: docker-compose.prod-run.yml [hunk 8]
```diff
@@ -3,12 +3,14 @@ version: "3.7"
 services:
   frontend:
     image: etimodanielwinther/diamonds2_frontend:${DIAMONDS_DOCKER_TAG}
+    restart: always
```
Comment (indrif) [inline docker-compose.prod-run.yml:6]: Dont know if this restarts container when docker is restarted. But please try!

File: auto-scale-aws.sh (hunk 1 above)
Comment (ForsbergPhilip) [reply auto-scale-aws.sh:3]: Yep, forgot to change this. Fixed now.

File: auto-scale-aws.sh (hunk 2 above)
Comment (ForsbergPhilip) [reply auto-scale-aws.sh:16]: Though of that aswell. Fixing this.

File: auto-scale-aws.sh (hunk 4 above)
Comment (ForsbergPhilip) [reply auto-scale-aws.sh:31]: Single equal sign works aswell. But changing to double to be more clear.

File: auto-scale-aws.sh (hunk 5 above)
Comment (ForsbergPhilip) [reply auto-scale-aws.sh]: Fixed!

File: auto-scale-aws.sh (hunk 6 above)
Comment (ForsbergPhilip) [reply auto-scale-aws.sh]: Fixed!

File: auto-scale-aws.sh (hunk 7 above)
Comment (ForsbergPhilip) [reply auto-scale-aws.sh]: Fixed!

File: docker-compose.prod-run.yml (hunk 8 above)
Comment (ForsbergPhilip) [reply docker-compose.prod-run.yml:6]: Will do :)

**Pull request merged**